R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -2083,6 +2083,46 @@ test_that("dapply() on a DataFrame", {
   expect_identical(expected, result)
 })
 
+test_that("gapply() on a DataFrame", {
+  df <- createDataFrame (
+    sqlContext,
+    list(list(1L, 1, "1", 0.1), list(1L, 2, "2", 0.2), list(3L, 3, "3", 0.3)),
+    c("a", "b", "c", "d"))
+  expected <- collect(df)
+  df1 <- gapply(df, function(x) { x }, schema(df), df$"a")
+  actual <- collect(df1)
+  expect_identical(actual, expected)
+
+  # Computes the sum of second column by grouping on the first column and checks
+  # if the sum is larger than 2
+  schema <- structType(structField("a", "integer"), structField("e", "boolean"))
+  df2 <- gapply(
+    df,
+    function(x) {
+      y <- data.frame(x$a[1], sum(x$b) > 2)
```

Comment:
Does this require that `x` is order by `a`?

If not, should gapply() could be done by   dapply(repartition(df, "a"), f) ?